pkg/router/router.go

```diff
@@ -294,24 +294,34 @@ func (r *router) Serve(ctx context.Context) error {
 }
 
 func (r *router) serveTransportManager(ctx context.Context) {
+	done := make(chan struct{}, 1)
```

Comment:
What do you think about accompanying this with `once := new(sync.Once)` and call `once.Do(func(){close(done)})`? Just in case we encounter more than two `handleTransportPacket` go routines that reach `err == transport.ErrNotServing`. I'm assuming that in this case, one will hang.

Maybe another option is to have the `<-done:` case trigger a loop that empties `done`.

Reply:
I know that `serveTransportManager` will only be called once, but I feel we should still maintain a clean closing logic.

Reply:
@evanlinjin This sounds like a good idea, thanks.